openedx/core/lib/gating/api.py

```diff
@@ -299,3 +305,143 @@ def get_gated_content(course, user):
                 {'id': user.id}
             )
         ]
+
+
+def is_gate_fulfilled(course_key, gating_content_key, user_id):
+    """
+    Determines if a prerequiste section specified by gating_content_key
+    has any unfulfilled milestones.
+
+    Arguments:
+        course_key (CourseUsageLocator): Course locator
+        gating_content_key (BlockUsageLocator): The locator for the section content
+        user_id: The id of the user
+
+    Returns:
+        Returns True if section has no unfufilled milestones or is not a prerequiste.
+        Returns False otherwise
+    """
+    gating_milestone = get_gating_milestone(course_key, gating_content_key, "fulfills")
+    if not gating_milestone:
+        return True
+
+    unfulfilled_milestones = [
+        m['content_id'] for m in find_gating_milestones(
+            course_key,
+            None,
+            'requires',
+            {'id': user_id}
+        ) if m['namespace'] == gating_milestone['namespace']
+    ]
+    return not unfulfilled_milestones
+
+
+def compute_is_prereq_met(content_id, user_id, recalc_on_unmet=False):
+    """
+    Returns true if the prequiste has been met for a given milestone.
+    Will recalculate the subsection grade if specified and prereq unmet
+
+    Arguments:
+        content_id (BlockUsageLocator): BlockUsageLocator for the content
+        user_id: The id of the user
+        recalc_on_unmet: Recalculate the grade if prereq has not yet been met
+
+    Returns:
+        tuple: True|False,
+        prereq_meta_info = { 'url': prereq_url|None, 'display_name': prereq_name|None}
+    """
+    course_id = content_id.course_key
```

Comment:
Nit on convention: please use `course_id` when we're passing around a string version and `course_key` when the object is a `CourseKey` object. That's the convention we aim for, even if we're not very consistent about it in the code yet.

Reply:
@ormsbee 
fixed